### ISSUE TITLE
Access to Keepass database on Googledrive is blocked 

### ISSUE THREAD

**Issue by Ramykee** [keeweb/keeweb#2191]
Hi,
I am trying to open my KeePass database (saved in Google Drive) from the web app (KeeWeb), but it does not work. I receive this message:
"This app is blocked

This app tried to access sensitive info in your Google Account. To keep your account safe, Google blocked this access."

It is a new Google account and I have not found anything in the settings to grant access.

Have you already a solution for this problem? is the app from google blocked?

**Comment by Aetherinox** [comment]
Are you referring to our official web app at: https://app.keeweb.info/?

Interesting enough, I just tried Google Drive integration and I was able to sign right in and select my vault. Going to run a few more tests, as maybe it's something directly related to the account in general.

Do you have a secondary google account you can try? And also a different browser? I was able to sign in using Edge, as well as Firefox.

**Comment by LarsGreen** [comment]
Hi,
I have a similiar issue, when trying to save my keeweb credentials to gdrive, I get an error message "This app is blocked".
I am using google chrome and the ubuntu application of keeweb.

The google account is around 2 or 3 years old.

Thanks for support!

**Comment by fahrenheit451fra** [comment]
Hello,

I've had the same problem for some time: ‘This application is blocked’ on two kdbx files saved on my google drive.
Impossible to save a new password (fortunately it still works to see them).

![SCR-20240906-jopw](https://github.com/user-attachments/assets/2fbc7b50-0d5a-481b-b028-87a2e300569e)

I generally use Mac (Sonoma 14.6.1) and Arc browser (I still use https://app.keeweb.info/). But whatever the machine or browser, if I haven't logged in once before this problem on the machine on which I want to see the passwords, I can't access them...

I hope you'll find a solution!

Thanks for this project!

**Comment by Aetherinox** [comment]
@fahrenheit451fra Yeah this makes sense. This is all related to the integration process with Google, and unfortunately they've changed their application standards so we're having to go through a bit more of a tightened control now when it comes to our app getting permission again to use the Google api unrestricted.

We'll get there.

**Comment by LostInSwiss** [comment]
same issue here. Strange thing is only happens when setting up keeweb on new computers, computers that are already connected seem to be ok.

**Comment by davinkevin** [comment]
+1, and unfortunately I just reset my phone 😭…

**Comment by NeelavaChatterjee** [comment]
+1

**Comment by Vo7kov** [comment]
+1

**Comment by Aetherinox** [comment]
Appreciate responses. I've done a bit more testing with the current public code and can confirm that if you've used the functionality in the past, you'll still be able to access your vault.

However, after the changes to Google, new users will have the issue.

I'm working through the Google requirements right now as I wrap up the code I'm working on that I can push to the repo, then we can start pushing forward.

**Comment by awvarya** [comment]
Hey! Any updates on that issue?

**Comment by Aetherinox** [comment]
Not directly yet. I'm mid-finishing up the remaining other stuff that needs to be wrapped up. I'm at a point right now that if I just stop, it's going to be a flat out mess to return and finish it up. But it's definitely high on my list.

**Comment by mbalc** [comment]
what's going on, what's the status? it's critical for me personally that this still doesn't work, should I look elsewhere for other software for this?

**Comment by juggler31** [comment]
+1

**Comment by Guillemsc** [comment]
+1

**Comment by Red-M** [comment]
This is because Google has decided that unfiltered drive access isn't a "good idea" and that applications that require it should be using the `https://www.googleapis.com/auth/drive.file` scope instead of https://github.com/keeweb/keeweb/blob/master/app/scripts/storage/impl/storage-gdrive.js#L302

The [tier 3 CASA assessments](https://github.com/keeweb/keeweb/issues/2190#issuecomment-2262104922) are ongoing security audits that cost a lot of money and have to be completed by Google's selected 3rd-party. Another example of this is https://www.youtube.com/watch?v=UMiB5Z7n6Y8

The entire assessment is a waste of time and money for the developers here and in my opinion, would be better suited to switch to the other drive scope.

**Comment by Nu11u5** [comment]
The block suddenly hit me today - unsure why it took this long, since I had to approve OAuth regularly before.

@Aetherinox have you looked at using the drive.file scope and the Google Picker API to allow file selection?  This would have to replace your built-in file browser, or rather it would only be able to show files granted "ownership" to the application by the Google Picker API.

**Comment by Arrorn** [comment]
Yeah, the block hit me mid-day today. It was previously working, but I had to switch to using the file synced via the Google Drive application.

**Comment by BrixSat** [comment]
> what's going on, what's the status? it's critical for me personally that this still doesn't work, should I look elsewhere for other software for this?

Such a lame comment. This is the best software around. We should thank the effort and love employed freely to help the world and not rant about it.

Thanks to the creator.

**Comment by rarunkarthik** [comment]
This happened to me too today, was working all these days with my already existing acct. I did not change the acct.

**Comment by Aetherinox** [comment]
> The block suddenly hit me today - unsure why it took this long, since I had to approve OAuth regularly before.
> 
> @Aetherinox have you looked at using the drive.file scope and the Google Picker API to allow file selection? This would have to replace your built-in file browser, or rather your file browser would only be able to show files granted "ownership" to the application by the Google Picker API.
> 
> https://developers.google.com/drive/picker/guides/overview

Yes, this was looked into by me last week. Right now I'm putting everything together that was done in the next update, and then I'll see where the status is at that time and decide which route to take.

At a minimum, the rest of the repository needs to be updated so that at least KeeWeb is re-buildable by the general public. Because if that were an option right now, people could just get their own API Key (free) and pop it in and use it as a temp solution. I forget the number (I think it's 100), there's no restrictions.

KeeWeb in general has to have an approved application because our API is used for all users who download and use the app.

**Comment by mduft** [comment]
Hey. Can confirm that I was unable to setup a new account a few weeks back for my daughter and decided to just wait a little for this to be resolved. However, I unfortunately can *also* confirm that as of today I was logged out of my existing acounts and cannot log back in. A rapid solution would be... very much appreciated :D

**Comment by abolychev** [comment]
The block suddenly hit me today.

**Comment by Red-M** [comment]
> I forget the number (I think it's 100), there's no restrictions.

Yeah, theres up to 100 users with the `https://www.googleapis.com/auth/drive.file` scope

**Comment by vanceism7** [comment]
> This is because Google has decided that unfiltered drive access isn't a "good idea" and that applications that require it should be using the `https://www.googleapis.com/auth/drive.file` scope instead of https://github.com/keeweb/keeweb/blob/master/app/scripts/storage/impl/storage-gdrive.js#L302
> 
> The [tier 3 CASA assessments](https://github.com/keeweb/keeweb/issues/2190#issuecomment-2262104922) are ongoing security audits that cost a lot of money and have to be completed by Google's selected 3rd-party. Another example of this is https://www.youtube.com/watch?v=UMiB5Z7n6Y8
> 
> The entire assessment is a waste of time and money for the developers here and in my opinion, would be better suited to switch to the other drive scope.

Is there anything else that needs to be done here to remedy this issue besides replacing the google drive scope with `https://www.googleapis.com/auth/drive.file`? - Otherwise I assume someone would've made a PR for this already..?

**Comment by Nu11u5** [comment]
> Is there anything else that needs to be done here to remedy this issue besides replacing the google drive scope

If you use the drive.files scope, then only Drive files that the app creates or "owns" can be listed and accessed.  The method for assigning ownership to a Drive file is using the Google Picker API, so this would have to be added to the code.  The Google Picker API has it's own UI elements that must be displayed in a browser view from an origin URL that is trusted by the GCP project.

**Comment by Nu11u5** [comment]
> KeeWeb in general has to have an approved application because our API is used for all users who download and use the app.

This is only "brand verification", to prove you are not impersonating another app and own the associated domains, right?

I don't believe using the drive.file scope would trigger the requirement for "sensitive scope verification" or "restricted scope verification" that involves a more involved process.

**Comment by znarkd** [comment]
I borrowed some of the KeeWeb techniques in developing my own Google Drive interface.  My app uses the https://www.googleapis.com/auth/drive scope along with the Google Picker API.  While my "unverified" app is initially blocked, Google displays an override that allows for access.  It isn't the total block that KeeWeb is experiencing.

**Comment by davinkevin** [comment]
Both authenticated instance I had got blocked too today. 

Any solution to self-host keeweb?

**Comment by vanceism7** [comment]
> Any solution to self-host keeweb and still use GDrive? Or this will be subject to the same issue?

Pretty sure this will be subject to the same issue.
Unless you create your own Google API project. Then you could list your own email as the "test user" and continue to use it in test mode

**Comment by sergey-koba-mobidev** [comment]
This fix helped me as a temp solution https://github.com/keeweb/keeweb/issues/2190#issuecomment-2334603996

**Comment by devth8** [comment]
If I understood a solution is being researching for the application https://app.keeweb.info/?
Thank's

**Comment by vanceism7** [comment]
> If I understood a solution is being researching for the application https://app.keeweb.info/? Thank's

Yes, along with the electron desktop app as well

**Comment by vanceism7** [comment]
Ok guys, an update for all the peeps. I've been playing with the code testing things out, and from what I'm seeing, all we need to do is update the scope to `drive.file`. I just created my own google api project with drive.file scope, and then changed the scope in keeweb to `drive.file` and its working fine. 

The only minor annoyance is keeweb won't have access to the file in gdrive anymore. To fix this, all you need to do is download the file off of google drive, open it with keeweb, and then resync it to your google account - after this, keeweb can see the file fine.

So yea, I don't know who has access to the official keeweb google api, but we fix this in google and in the code, and then push a release and this is fixed.

**Comment by vasekp** [comment]
@vanceism7 If I use KeeWeb on several machines, won't this give access to just the instance that I "resynced" from?

**Comment by vanceism7** [comment]
> If I use KeeWeb on several machines, won't this give access to just the one that I "resynced" from?

@vasekp - No I don't believe so. I haven't tested this, but as far as I understand, the ownership of a file is assigned based on the app's Google API Client ID, which would be the same for all instances of Keeweb, so using it on multiple machines shouldn't be an issue

**Comment by elliot710** [comment]
So I have this issue as well. A couple months back my gf was experiencing the same problem while I was not. Not I have the same issue.

**Comment by pmithrandir** [comment]
@vanceism7 Thanks for the PR.

Will this solution allow other app to access the keepass file ?

On windows and linux, I'm using keeweb obviously, but on my mobile I'm using keepass as I don't beleive keeweb was made available.(neither have the autocomplete user / password keyboard)

Also, having the file in gdrive allow my wife to access it using her google account to ensure she gets access to the family password.

Will it be possible ?

Regards,
Pierre

**Comment by vanceism7** [comment]
@pmithrandir 

Those are good questions. I use the app KeepassDX on my android and was able to access the new file without any issues, but this may depend on what app you're using. 

I haven't tested accessing it with _separate_ Google account, so the jury is still out on that one. How do you have that setup? Did you explicitly share the database file with the other Google account? Or is it in some kind of shared folder? Maybe it doesn't matter, just curious. I'll give it a try today though to see what happens

**Comment by nikosandreou** [comment]
Same issue here as of today on my desktop app. I am using KeepassDX on Android and syncs without issues.

**Comment by LostInSwiss** [comment]
hi all. just to note there is a workaround here. just tried it and it works: https://github.com/keeweb/keeweb/issues/2190

**Comment by sibiantony** [comment]
For me the workaround in #2190 didn't work. I'm on linux desktop with electron built binary, and the console never opens. In the end, I did the build myself. All it requires is build with the changed `scope` and there was no need to create project/change things with Google console.

The build itself is broken at many places, so I thought of pushing the whole thing as a dockerized build for the community. For those on desktop linux using the Electron app, [this repo](https://github.com/sibiantony/keeweb-desktop-linux-build) could be useful.

* The build clones from @vanceism7's repo (can be changed in `Dockerfile`)
* `electron-builder` is fixed to version `22.11.4` for compatibility reasons
* Overrides the default `Gruntfile` for building only the AppImage binary. (This also builds the deb package which is not tested)
* The rpm build is broken and is disabled along with snap builds.
* The built binaries can be found in `dist`.

That said, I prefer the ongoing PRs get merged, and have official built binaries in a release.

**Comment by leorochael** [comment]
@vanceism7 
>The only minor annoyance is keeweb won't have access to the file in gdrive anymore. To fix this, all you need to do is download the file off of google drive, open it with keeweb, and then resync it to your google account - after this, keeweb can see the file fine.

Losing access to the file might be unavoidable, but downloading off of google drive might not be necessary to restore access.

Perhaps access can be granted/restored by selecting the file with the official [Google Drive Picker](https://developers.google.com/drive/api/guides/picker), instead of trying to navigate to the file directly using `https://www.googleapis.com/auth/drive` scope APIs.

Cc: @antelle .

**Comment by perepicornell** [comment]
Hi everyone!

For a temporary workaround until a new version that fixes that is released, I wanted to synchronize the Google Drive folder locally. Very easy to do [using rclone](https://ucr-research-computing.github.io/Knowledge_Base/how_to_mount_google_drive.html) in Linux.

But then when opening the local file in KeeWeb, it says that the app will load the file internally and not write on it again, unless I export it.

So this workaround doesn't work for me because I can't relay on remembering to export the file on every edition I do. I assumed that the app will be working against the local file...

I thought other might be interested.

Thanks for all the efforts, either to the contributors and to the community posting workaround and ideas.

**Comment by gsportelli** [comment]
> But then when opening the local file in KeeWeb, it says that the app will load the file internally and not write on it again, unless I export it.

I think that the message is misleading. I've tried and if you select a local file, then synchronise as you would with Google Drive, then it merges the changes into the file as it would with Google Drive.

**Comment by Bogghi** [comment]
I've got a Vault synched across all my devices with synchthing and on each I have installedd keeweb, with local open on macos seems to work just fine. On android i open the file with KeepassDBX. 

I've found this to be the best solution for me.

**Comment by perepicornell** [comment]
> > But then when opening the local file in KeeWeb, it says that the app will load the file internally and not write on it again, unless I export it.
> 
> I think that the message is misleading. I've tried and if you select a local file, then synchronise as you would with Google Drive, then it merges the changes into the file as it would with Google Drive.
> 
> If confirmed, this should solve the problem for many.

I tried it before with a co-worker who is not having this "blocked by Google" error and he didn't see a change that I did, after clicking synchronize, while having the Drive folder mounted locally with rclone.

It was just a quick test so we might've done something wrong.

**Comment by opti93** [comment]
I switched to oneDrive its working like a charm

**Comment by carlbennett** [comment]
> I switched to oneDrive its working like a charm

Dropbox is working great as well. A feature of KeePass/KeeWeb, store and load the database to and from wherever you want, on any or most devices. If gdrive isn't working–try somewhere else!

I am thinking of using [KeeWeb's WebDAV](https://github.com/keeweb/keeweb/wiki/WebDAV%20Config) via the [nginx dav module](https://nginx.org/en/docs/http/ngx_http_dav_module.html), seems like the prudent choice. A cloud storage vendor may disable the API or alter the deal at any time, rendering KeeWeb incompatible. Self-hosted WebDAV with HTTP auth, IP allowlist, and CORS ACL header(s) puts me back in control, without worrying about it being unavailable one day like gdrive did to me here. I already have an nginx server I can use for this, so it's not too inconvenient to setup and secure.

Thank you for the open-source code and I'm subscribing to know of when the gdrive permission fix becomes widely available. 👍🏻

**Comment by leorochael** [comment]
@vanceism7 
>The only minor annoyance is keeweb won't have access to the file in gdrive anymore. To fix this, all you need to do is download the file off of google drive, open it with keeweb, and then resync it to your google account - after this, keeweb can see the file fine.

An alternative to implementing the full GDrive picker integration would be to provide ["Open With" integration](https://developers.google.com/drive/api/guides/integrate-open).

Though it could be about as complex as using the GDrive Picker.

**Comment by andre-m-faria** [comment]
> > I switched to oneDrive its working like a charm
> 
> Dropbox is working great as well. A feature of KeePass/KeeWeb, store and load the database to and from wherever you want, on any or most devices. If gdrive isn't working–try somewhere else!
> 
> I am thinking of using [KeeWeb's WebDAV](https://github.com/keeweb/keeweb/wiki/WebDAV%20Config) via the [nginx dav module](https://nginx.org/en/docs/http/ngx_http_dav_module.html), seems like the prudent choice. A cloud storage vendor may disable the API or alter the deal at any time, rendering KeeWeb incompatible. Self-hosted WebDAV with HTTP auth, IP allowlist, and CORS ACL header(s) puts me back in control, without worrying about it being unavailable one day like gdrive did to me here. I already have an nginx server I can use for this, so it's not too inconvenient to setup and secure.
> 
> Thank you for the open-source code and I'm subscribing to know of when the gdrive permission fix becomes widely available. 👍🏻

Last week I was thinking the same thing about WebDAV, your comment just gave me more reasons to make the move.

**Comment by mbalc** [comment]
possibly related, but possibly not? - same problem was happening for people trying to sign in in KDE to Google; a workaround was described here, involving a change of scope for Google Drive https://discuss.kde.org/t/kde-online-accounts-not-signing-in/3411/38

**Comment by vanceism7** [comment]
> possibly related (but maybe not?) - same problem was happening for people trying to sign in in KDE to Google
> 
> a temporary workaround that I'm actually using successfully is described here, involving a change of scope for Google Drive https://discuss.kde.org/t/kde-online-accounts-not-signing-in/3411/38

Sounds like they arrived at a different conclusion by using a different client id that had possibly already passed verification.

But there's already a PR open which reduces the scope from `drive` to `drive.file` - we're just waiting for it to be merged at this point

**Comment by alexcodemaster** [comment]
@Aetherinox can you please check this PR to fix this issue? 
@vanceism7  thank you very much!

**Comment by WangGuan11** [comment]
Is it still okay? Very urgent

**Comment by SebastienE** [comment]
Have the same problem.
Do you have news about fix ?

**Comment by vanceism7** [comment]
We should have a new release coming out any day now. 

See https://github.com/keeweb/keeweb/pull/2208#issuecomment-2526357951

**Comment by lal12** [comment]
For anyone looking for a quick fix:
When the browser opens a new window asking for the permission and asking you for selecting an account, you have to change the url slightly. It's a long URL but somewhere in there is the text "%2Fdrive", insert ".file" after this, so it says "%2Fdrive.file", then press enter. Now continue as usual.

**Comment by andre-m-faria** [comment]
> For anyone looking for a quick fix: When the browser opens a new window asking for the permission and asking you for selecting an account, you have to change the url slightly. It's a long URL but somewhere in there is the text "%2Fdrive", insert ".file" after this, so it says "%2Fdrive.file", then press enter. Now continue as usual.

`%2Fgdrive.html`? I changed it to `%2Fgdrive.file.html` but this didn't worked.

**Comment by lal12** [comment]
> > For anyone looking for a quick fix: When the browser opens a new window asking for the permission and asking you for selecting an account, you have to change the url slightly. It's a long URL but somewhere in there is the text "%2Fdrive", insert ".file" after this, so it says "%2Fdrive.file", then press enter. Now continue as usual.
> 
> `%2Fgdrive.html`? I changed it to `%2Fgdrive.file.html` but this didn't worked.

No, sorry it was ambiguous, I updated my comment.

**Comment by vanceism7** [comment]
@lal12 - Ah you're right! That totally worked! I didn't realize you could just manually change the scope and google would happily accept it - it feels a bit hacky, but its a great workaround in the meantime

**Comment by andre-m-faria** [comment]
> > > For anyone looking for a quick fix: When the browser opens a new window asking for the permission and asking you for selecting an account, you have to change the url slightly. It's a long URL but somewhere in there is the text "%2Fdrive", insert ".file" after this, so it says "%2Fdrive.file", then press enter. Now continue as usual.
> > 
> > 
> > `%2Fgdrive.html`? I changed it to `%2Fgdrive.file.html` but this didn't worked.
> 
> No, sorry it was ambiguous, I updated my comment.

Oh, I was doing it wrong too. Thank you very much is working.

**Comment by vanceism7** [comment]
Just an update for everyone here: https://github.com/keeweb/keeweb/pull/2208 was merged a week ago or so, so we should be seeing this fixed once a new release of the app is made.

**Comment by Aetherinox** [comment]
An update to his update. His change was also migrated to the branch https://github.com/keeweb/keeweb/tree/develop/1.19.0

This is where I am doing all the work on the newest version. So if you guys want to build and use the newest copy, you're more than welcome to.

Building keeweb now should result in no errors. I even enhanced the README to explain things more.

**Comment by Aetherinox** [comment]
Also, I've spent the last 30 or so hours trying to bust tail to get things done.

I've finished up the new docker containers, all available on https://github.com/keeweb/keeweb/pkgs/container/keeweb.

Those docker containers also have the fix for this drive issue that was submitted in the PR.

Another user also wants an arm64 compatible version of Keeweb, and since that's going to take a bit of time, seeing as we have more pressing issues to address first like getting integration up to 100% again, and Electron, I've at least put out a container that can run on a arm64 platform behind docker.

There's also a new development version

Along with all of that, I've spent the time to write extensive documentation for the new docker builds, which includes Traefik integration, as well as Authentik integration. Those docs are on https://github.com/keeweb/keeweb/tree/docker/keeweb

The windows build works, compiles great. Next I need to test the Linux build, and somehow.... test macos. That's the only royal pain is that I don't have a mac, so I'm driving blind.

**Comment by Aetherinox** [comment]
> How could I help testing on macOS?

Once I push out the final build files, it would be a matter of just cloning the v1.19.0 repo to your local machine, and then running the command to build on mac, off the top of my head, I believe is

```
npm run dev-desktop-macos
```

While it would help, longer-term it would be a pain for certain things, because I can't just expect people to sit at the computer all day during my work hours, me push changes, have them re-pull, re-build, test, and then somehow communicate to me what is going on if an issue arises.

So for now I'm going to finish the files needed for the build, and then some up with some type of solution. Testers will be good to test on various different machines, but from a debugging stand-point of me having to see the symptoms of an issue, it's just not realistic. Appreciate the offer though.

**Comment by gesellix** [comment]
Maybe some aspects can be checked with automated tests, then running on GitHub's macos-runner. This won't ensure 100% coverage, but I guess it would help reduce your and other crazy people's time to prevent regressions :)

**Comment by Aetherinox** [comment]
> Maybe some aspects can be checked with automated tests, then running on GitHub's macos-runner. This won't ensure 100% coverage, but I guess it would help reduce your and other crazy people's time to prevent regressions :)

That's actually what I've been slowly doing, is introducing additional coverage / tests. But realistically, only about 25% of Keeweb's code is covered right now. And I can't spend exorbitant amounts of time on that when there's other more important things. But it's definitely something on my list. It'll make my life a lot easier. Right now I'm just throwing new tests in as I work on a feature related to that test. Slowly.

**Comment by pmithrandir** [comment]
HelloThere is some service online that let you use a Mac per hour.Not sure how much it is...PierreSent from Android deviceLe 19 déc. 2024 11:05, Aetherinox ***@***.***> a écrit :

How could I help testing on macOS?

Once I push out the final build files, it would be a matter of just cloning the v1.19.0 repo to your local machine, and then running the command to build on mac, off the top of my head, I believe is
npm install
npm run dev-desktop-macos

While it would help, longer-term it would be a pain for certain things, because I can't just expect people to sit at the computer all day during my work hours, me push changes, have them re-pull, re-build, test, and then somehow communicate to me what is going on if an issue arises.
So for now I'm going to finish the files needed for the build, and then some up with some type of solution. Testers will be good to test on various different machines, but from a debugging stand-point of me having to see the symptoms of an issue, it's just not realistic. Appreciate the offer though.

**Comment by bikefanatic16** [comment]
Hey everyone, I'm sorry 'cause I'm sure it's a dumb question but I have absolutely no experience with this. I'm using keeweb on my mobile phone and my desktop pc as a browser app connected with a pwdb on my google drive. As we all experienced, the synchronisation on the gdrive doesn't work anymore and I understood, that Aetherinox is still working on this issue in developing v1.19 which is allready in a testing phase. But what I don't understand is: do I have something to do to update my "app" or do I only have to wait, until v1.19 is released an the browser app will update automatically? And: is there a rough time line, when v1.19 will be released? Thanks for your help everyone.

**Comment by znarkd** [comment]
This bug took a good number of users offline.  I'm greatful that Aetherinox stepped up to fix the problem.  However, I am frustrated that what seems to be a relatively simple fix has been tied together with a release of new features.

**Comment by rpaech** [comment]
Remember chaps that this is open-source software; no one is getting paid to update or fix it.  If the guys working on this believe that it is more efficient use of their free time to combine the fix with the next release then that is their right.  If you use open-source software, you need to accept the risk that things may break and take time to resolve.

**Comment by nikosandreou** [comment]
> Remember chaps that this is open-source software; no one is getting paid to update or fix it. If the guys working on this believe that it is more efficient use of their free time to combine the fix with the next release then that is their right. If you use open-source software, you need to accept the risk that things may break and take time to resolve.

Sorry but I totally disagree with that. If the fix is a simple one and can go through a PR easily, you release the fix even as a minor release (1.18.8). This is a hotfix and should be released ASAP. Then if you want to add new features, add them to a major release. T

**Comment by Aetherinox** [comment]
@nikosandreou If this were a normal release, then I'd agree with the time. But in this instance, it's not that simple.

1. The app integrations were broken, which originally the plan (instead of down-grading the permissions), was to correct the issue. 

2. Google Drive is far from the only thing that broke (also the browser plugin for certain browsers) since the last 2021 release. And those were focused on first. Including the ability to now build the package without issue, and not release jut a patch which ships all those vulnerabilities along with it from the outdated packages.
   1.  This includes people who had bugs with OneDrive, so is their issues not as important? Because a lot of these issues can equally break a certain part of the app just as well.
   2. Anything outside of actual fixes has been a few hours here and there adding a few new things users can ta least anticipate looking forward to after a 3+ year gap, and that is while waiting on other things to finish that handicapped the ability to work on those items at that time.
   3. Including setting up new apps for Microsoft / Google, since Ant keeps the original accounts, and I have no clue if they are tied into personal stuff or not.

4. Ant originally signed the binaries with trusted certs, which means that for the Windows binary as an example, you don't get the popup screen when the app is launched telling you that it is an untrusted source. We've had to take time to get all new certs verified so that the binaries can be signed normally.  While not bugging Ant, because he's been going through personal stuff, and I'm not up to helping manage this repo, and also bugging him. He hasn't released for a reason, and I'm not forcing him back.
   1. This also includes not only the Windows binary working, but also that the Mac and Linux ones do as well and have not been broken by the dependency updates. All three have to be in working order, not just dropping different releases of different binaries as I fix them. Windows binaries work, Linux works, and the remaining is mac.

5. Others are perfectly aware of the state of Keeweb, so I've made it plainly blunt from the beginning that if others knew how to code, and also wanted to get their hands dirty and help get some of this stuff completed, by all means. I'll review PRs all day if they address the issues.

6. Minus the binary certs, I've ensured that the existing copy is at least build-able by people now, including new [docker images](https://github.com/keeweb/keeweb/pkgs/container/keeweb), and the [web app](https://beta.keeweb.info/). So people at least have a few routes to using the updated code, instead of no way. Both of those links include the new fixes, and anything else after it is finished and tested. Including the Google Drive fix.

Had this been just a few month gap and we weren't really playing catch-up, I could see just pushing hot-fixes, but the state was far different.

Source is also buildable now, two commands for web-app, three for windows.

**Comment by bikefanatic16** [comment]
Well that didn't went out as intended. I absolutely didn't want to bring pressure on @Aetherinox or complain about the time line because I'm very aware of the fact that this is a spare time project and I appreciate the work on that very highly. 

I just wanted to understand if and what I have to do, when the release is finished.
Maybe someone else can answer my questions.

**Comment by rpaech** [comment]
> I just wanted to understand if and what I have to do, when the release is finished. Maybe someone else can answer my questions.

Sorry @bikefanatic16, I wasn't originally responding to you; though I wasn't any help either.  Hopefully the following answers your question...

If you want to be notified when the next release comes out, go to the main page for the KeeWeb repo in GitHub and, near the top right of the page, click on the **Watch** drop-down button; then, from the list the comes up, click **Custom** and select the **Releases** checkbox.  GitHub should then send you an email when the update is released.

Assuming that updates to the app are pushed out like they used to, within the KeeWeb app there was usually an icon that showed up in the bottom status bar to indicate that an update was available.  To perform the update, you had to go into **Settings** and under the **General Settings** section select the **Check for Updates** button.  That said, it might be a bit more complicated given how the original certificates need to be replaced.  If that's the case, you will probably need to manually download and reinstall the app.

**Comment by Aetherinox** [comment]
@bikefanatic16 sorry if that came off in any way aggressive. Was just outlining the status. There's been a load of work to handle.

> I just wanted to understand if and what I have to do, when the release is finished.

Well it depends on which platform you are using or what you're willing to use.

If you want to use the Windows binaries:

```
git clone https://github.com/keeweb/keeweb.git -b develop/1.19.0 .
```

That will download the source files, the only downside is ensuring you've installed Node on your system. There's a Windows binary for Node, so it's an easy install.

Then `cd` into the folder with the files you downloaded and run

```
npm install
```

That will take a few moments, and after that is complete, you can do one of the following:

### Build Test

```
npm run dev
```

That will build and set up the webserver for you automatically. You just need to open your browser and go to:
- http://localhost:8085/

### Build Windows Version

```
npm run dev-desktop-windows
```

That will give you the actual executable in `\tmp\desktop\KeeWeb-win32-x64\KeeWeb.exe`

### Docker

For the docker version, you don't need to do any of the above, if you've done docker pulls before, it's just a matter of creating a `docker-compose.yml` and adding the new image

```yml
services:
    keeweb:
        container_name: keeweb
        image: ghcr.io/keeweb/keeweb:latest
```

If you need detailed instructions, I wrote up a pretty big readme for the docker image:
- https://github.com/keeweb/keeweb/tree/docker/keeweb

In terms of not having to jump these hoops, and getting a working Windows / Linux build. Windows is ready. I need to test Linux which I plan on doing this week, and that leaves the MacOS version; which I'm currently working on a solution to test with (I don't have mac).

Once this initial stuff is done, then releases can go back to normal. Patches / hotfixes, and then major releases. I can push any time I want.

Then the last order of business, is the certificates for signing.

Google just messaged back tonight oddly enough, and we completed the initial app, it's pending review now. So there's light at the end of this tunnel.

---

As far as notifications of a release:

![rJBrsSxPLt](https://github.com/user-attachments/assets/4627a1bf-527c-4184-9b6b-a871cccb86ea)

---

@rpaech Yeah, the update notifications work. There was a bug a few weeks ago, but I've fixed that since then. And I plan on doing releases in the same fashion as Ant. The certs were actually one of the issues.

**Comment by znarkd** [comment]
The "Downgrade gdrive scope to drive.file" fix implemented on beta.keeweb.info isn't working for me.  While I'm able to access my Google Drive account, no files are listed.  I only see the Shared drives folder and the Shared with me folder.  If I select the Shared drives folder, I get a http status 403 error.

![Screenshot 2025-01-02 213801](https://github.com/user-attachments/assets/a122fe9c-dddc-4c79-af77-f1d15d1346cc)

**Comment by alanbirtles** [comment]
> The "Downgrade gdrive scope to drive.file" fix implemented on beta.keeweb.info isn't working for me. While I'm able to access my Google Drive account, no files are listed. I only see the Shared drives folder and the Shared with me folder. If I select the Shared drives folder, I get a http status 403 error.
> 
> ![Screenshot 2025-01-02 213801](https://private-user-images.githubusercontent.com/5201266/399869837-a122fe9c-dddc-4c79-af77-f1d15d1346cc.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4OTQ1NzIsIm5iZiI6MTczNTg5NDI3MiwicGF0aCI6Ii81MjAxMjY2LzM5OTg2OTgzNy1hMTIyZmU5Yy1kZGRjLTRjNzktYWY3Ny1mMWQxNWQxMzQ2Y2MuSlBHP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDNUMDg1MTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzBlNGMxNzMxYWE0ZDEwYjQ5OGNmYWQyNjRhYzA1OGRjNTg0OWE2YTM1MTRlNjI4MDk1ZWU4NjY5MjVjMTIyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.dyjyS2dLlaFkpQj626QLTASScbkvjal6YktO9rD4ZFE)

Looks like it only works for files in the root of your gdrive

**Comment by Aetherinox** [comment]
> Looks like it only works for files in the root of your gdrive

Yup. That's the downside to this fix, which is just a temp patch.

The best solution is going to be getting us re-verified with Google, because this patch can't stay like this. It means that anyone who had a kdbx in a subfolder, will no longer be able to access it and will have to manually move it.

That was partly my reservation with releasing a "fix" is because it's not really a fix. It'll bring on a separate set of users in here who can not access items at all. The old scope still allows existing users who had authenticated prior to still access their drive without error.

**Comment by leorochael** [comment]
> > Looks like it only works for files in the root of your gdrive
> 
> Yup. That's the downside to this fix, which is just a temp patch.
> 
> The best solution is going to be getting us re-verified with Google, because this patch can't stay like this. It means that anyone who had a kdbx in a subfolder, will no longer be able to access it and will have to manually move it.

As I mentioned [in the PR](https://github.com/keeweb/keeweb/pull/2208#issuecomment-2499258720) and elsewhere,

Perhaps access can be granted/restored by selecting the file using the official [Google Drive Picker](https://developers.google.com/drive/api/guides/picker), instead of trying to navigate to the file directly using https://www.googleapis.com/auth/drive scope APIs.

This might allow KeeWeb to keep using the `drive.file` scope while providing access to files inside any folder in Google Drive.

**Comment by grossetti** [comment]
Hi all,

I tried the beta webapp and it didn't work out for me too, but for
different reasons. I have two gdrive accounts with different key files. On
one account, my file is directly on the gdrive root and it wasn't visible
in Keeweb. On the other account, I have 2 key files directly on the gdrive
root, and only one of the files was visible (unfortunately not the one I
wanted).

Cheers,
Gab

**Comment by Aetherinox** [comment]
> Perhaps access can be granted/restored by selecting the file using the official [Google Drive Picker](https://developers.google.com/drive/api/guides/picker), instead of trying to navigate to the file directly using [googleapis.com/auth/drive](https://www.googleapis.com/auth/drive) scope APIs.

I'll dig into the docs and see if this would present any other issues with how the system is currently set up.  

We're currently pending review by Google, the `drive.file` scope requires no verification, but I'm unsure on how long it's going to take Google to get to our application.

**Comment by vanceism7** [comment]
I mentioned this before when I originally downgraded the scope to drive.file

**You must download the file manually and then resync it to your Google drive**

Drive.file only allows Keeweb to see files it's created. Any files created prior to the downgrade will not be "owned" by Keeweb, so Keeweb won't be able to see it.

Using the Google picker API will ultimately remedy this, but this is the workaround in the meantime

@znarkd @grossetti @alanbirtles @GamerClassN7

**Comment by Red-M** [comment]
With regards to getting re-verified by Google, you might want to re-read [my comment](https://github.com/keeweb/keeweb/issues/2191#issuecomment-2472132250)

**Comment by GamerClassN7** [comment]
Any news regarding this topic ?  I tried the latest beta and I am able to authenticate no error but I am unable to bound files from my drive just 2 folders my drive returned 405 or shared with me and there is nothing in it

**Comment by vanceism7** [comment]
@Aetherinox 

Seems like a lot of people are running into the same issue of Google drive not showing them files. I originally wrote how to resolve this and reiterated it in this comment: https://github.com/keeweb/keeweb/issues/2191#issuecomment-2569468133

The solution:

---

**You must**:
- Download the keepass file
- Open it with Keeweb
- Resync it to your Google drive via Keeweb

---

This issue is technically resolved so we should probably close it and lock the comments so the workaround doesn't get buried beneath more comments

If there are _other_ issues preventing people from accessing files besides empty folders, those would be better served with a new issue

**Comment by Aetherinox** [comment]
@vanceism7 I'll close this for now.

Do me a favor and create a new post with a summary of the issue and outlining these steps, and I'll pin it for now to the issues section.

I'm currently looking into the Google Picker API.

**Comment by bikefanatic16** [comment]
@vanceism7 
> I mentioned this before when I originally downgraded the scope to drive.file
> 
> **You must download the file manually and then resync it to your Google drive**
> 
 But this only works in the beta version, is this right?

**Comment by vanceism7** [comment]
@bikefanatic16 Yes - only with the beta version. 

Or you can achieve a similar fix on 18.7 using the workaround outlined here: https://github.com/keeweb/keeweb/issues/2191#issuecomment-2532114892

Which also downgrades the scope to drive.file, and then you still need to follow the steps of resyncing the file from my previous comment 
https://github.com/keeweb/keeweb/issues/2191#issuecomment-2571333316

@Aetherinox 
Sorry, just to clarify - you want me to open a new issue, something like "Google Drive shows no files in Keeweb"?

**Comment by bikefanatic16** [comment]
@vanceism7 ah great. This put the pieces together for me. First time I only tried to edit the link without down- and upload the file. On my desktopversion of the app its working for now. Now I just have to find out, how it will work on my mobile phone app. Thanks for your help!

**Comment by scottmcosta** [comment]
What steps should we take if we have a file that is shared with us from another google account? I have a file shared with my wife and she cannot "Resync it to your Google drive via Keeweb" with her google account because that would just create a separate copy of the file from the one I have shared with her.

**Comment by vanceism7** [comment]
@scottmcosta 

In general, resyncing the file essentially duplicates it.

I'm not sure if this will work, but if you Resync and then share it with her via Google drive, she might be able to access it on her own Keeweb.

I haven't tested this, but I'd say it's worth a shot

**Comment by scottmcosta** [comment]
@vanceism7 
Yep, that was the first thing I tried, since I had to resync to get it working on my own computer. The solution discussed above worked great for me. It didn't seem to work for my wife. When I went to open a file from Google drive on my wife's computer, KeeWeb was unable to find it when I clicked on the shared files option. ¯\\_(ツ)_/¯

**Comment by vanceism7** [comment]
Ah gotcha, that's unfortunate. I think for that use-case, we're more than likely stuck until the google picker api is implemented then. Sorry about that!

**Comment by werelord** [comment]
Any timeline on when 1.19 (beta or stable) will be released as a desktop app? I would love to get syncing back and would prefer to not attempt building this directly..

**Comment by SergiusTheBest** [comment]
@werelord I tried to build it for myself but have got an error on launching the resulting electron app. So I switched to One Drive instead of Google Disk. It was very easy to do and works flawlessly.

**Comment by agilgur5** [comment]
> Any timeline on when 1.19 (beta or stable) will be released as a desktop app? I would love to get syncing back and would prefer to not attempt building this directly..

For anyone else looking, you can use https://beta.keeweb.info in the interim temporarily until a stable version is available. It worked for me

**Comment by Aetherinox** [comment]
Yeah, I made some adjustments to the beta channel in an attempt to help users get around the issue. Should work as a temp.

**Comment by daalla** [comment]
> Yeah, I made some adjustments to the beta channel in an attempt to help users get around the issue. Should work as a temp.

hi @Aetherinox, thanks a lot for the beta release! is there a way to download it as a desktop app?

**Comment by Konkrad** [comment]
![Image](https://github.com/user-attachments/assets/caa9c84b-f5d9-404a-92e4-10204eb6f99f)

I have a question. Once all is done and deployed, will it mean that I can define the App to open `.kdbx` files automatically with the Web App?

**Comment by gavoja** [comment]
@Aetherinox 

Logging in to Google Drive works in beta, however the only available sources are "Shared Drives" and "Shared with me". It does not currently list any owned files.

![Image](https://github.com/user-attachments/assets/7a5f8d77-7a6c-4403-b96e-94869b888465)

**Comment by alanbirtles** [comment]
@gavoja see posts above

**Comment by Aetherinox** [comment]
@gavoja Someone posted a work-around above. There's a few topics on this subject, I believe its in this thread.

I'm currently speaking to Google about our approval. So this should not be a headache forever. The sooner the better.

If for some reason, the computer gods above decide that Google would not approve us for the needed scope, I have an alternative solution I'm prepared to implement, which would involve using another method of integrating Google instead of Google's method.

**Comment by Nu11u5** [comment]
> I'm currently speaking to Google about our approval. So this should not be a headache forever. The sooner the better.

This is to get full Drive API access or are you implementing the Google Picker API?

**Comment by mauroagr** [comment]
> I'm currently speaking to Google about our approval. So this should not be a headache forever. The sooner the better.

Hello,
Any news about this?

**Comment by devth8** [comment]
Do we have news ?

**Comment by tomasMizera** [comment]
The workaround is near impossible to use on mobile, is there any way we can help supporting the fix? :)

**Comment by ashelleyPurdue** [comment]
@Aetherinox

I understand that this is a project that you work on in your spare time, and I greatly appreciate that it has been given to us for free with no strings attached.  I normally don't demand or expect anything of open source maintainers.  But...

It's been 6 months since google drive syncing broke, and a workaround still has not been released for Desktop.  I've not been able to safely change any of my passwords for this entire time, since I don't want to risk the old ones overwriting the new ones once syncing is eventually fixed.

There is a PR that (supposedly) fixes this issue that has been open since last December.  It has just been sitting there this entire time.  A workaround has also been merged to the beta channel, but that channel only affects the website version, which I'm frankly unwilling to trust with all of my passwords.  I'm only willing to trust a Desktop app with that.

So, given that there are multiple tested workarounds, and given how urgent this issue is (and has been), why have you taken so long to pull the trigger?

Again, I understand that this is a free project that you've shared with us out of the goodness of your heart, **and I cannot thank you enough for that**.  It seems wrong to demand quick action from you, but I'm kind of in a bind now.  I need to change my expired banking password, but I can't do that safely while all of my passwords are in limbo.  I imagine others are in the same situation.  _Please_ push the workaround to the Desktop version and make a release.

**Comment by Aetherinox** [comment]
@ashelleyPurdue 

As said in another discussion (somewhere), the PR does not address all the issues with drive integration, it solves it for some, and continues to keep the problem for others. If you want to know if the fix actually works in your case, use our beta site https://beta.keeweb.info/ and see if you can open the vault. The push was migrated there until one of ~~two~~ three things happened.

1. Approval from Google which allows us to use the full scope;
2. Someone else proposed an alternative method which basically removes reliance on the Google Drive integration and uses a 3rd party library.
3. Use the scope we're allowed to access, and allow users to only be able to save to the root folder of Google Drive

My last reply to Google was 39 days ago, with no response yet to validate what I've submitted after they requested a few dependency updates where there were vulnerabilities.

In terms of wanting a release now. If it's that much of a need, since I've been dealing with internal family stuff for the last bit, I can either see what Google is going to do with this next request, or I'll let someone else work on the code and submit a proposed update to address it using the 3rd party library. Because right now it seems I'm the only one doing anything, and others have been more than free to step in and speed things up.

And also about the beta site comment, there's also a docker image with the temp proposed changes.

But at this moment, I can't possibly speed it up anymore than I am, so if someone else wants to step in, then I'll review the code once it's submitted since there needs to be someone here to try to manage it. Once I get a definitive answer on the Google scope, then I'll decide on what to do from there, or branch it and offer redundancy. 

Another note since I forgot before I left.

The app is also buildable, and you can register on Google for an API key and insert that, and then build the win32 since the API key is limited to you and isn't used by the entirely of the keeweb userbase. I fixed the build process quite a while ago, so it builds now without issue.

Next week once I return from dealing with internal family situation, I'm going to go through the remaining of what was proposed for 1.9 and decide from there.  Everything that needed to be fixed when the repo went stagnant has been fixed, it's just a matter of the Google Drive situation.

**Comment by Red-M** [comment]
A UI to put our own google API key would be nice.

**Comment by GamerClassN7** [comment]
> [@Aetherinox](https://github.com/Aetherinox)
> 
> Seems like a lot of people are running into the same issue of Google drive not showing them files. I originally wrote how to resolve this and reiterated it in this comment: [#2191 (comment)](https://github.com/keeweb/keeweb/issues/2191#issuecomment-2569468133)
> 
> The solution:
> 
> **You must**:
> 
> * Download the keepass file
> * Open it with Keeweb
> * Resync it to your Google drive via Keeweb
> 
> This issue is technically resolved so we should probably close it and lock the comments so the workaround doesn't get buried beneath more comments
> 
> If there are _other_ issues preventing people from accessing files besides empty folders, those would be better served with a new issue

No longer viable option i am unable to sasve local DB to GD:

![Image](https://github.com/user-attachments/assets/170869e3-f7a9-4f9c-b708-4f7a6e9b5e83)

**Comment by Nu11u5** [comment]
I've fixed this by implementing Google Picker in #2259.

**Comment by Aetherinox** [comment]
Let me review this, because I've also addressed the initial Google Drive issue, the only one remaining is a OneDrive bug people have reported. But I'd rather offer multiple solutions to avoid this in the future.  There was also a few WebDAV issues, but they are now corrected as well.

I do like this route better though in terms of not having to rely on 3rd party services to hold up development. Because dealing with Google's QA team can be a rather long process. Especially for the scopes we need.

**Comment by Aetherinox** [comment]
@Nu11u5 I got one of the major things I wanted to finish wrapped up. I just need to update the dockerfile and I'll start pushing to the repo again. 

I need to start wrapping up while I have time, and get a release going. Aside from the integration issues that were broken in the app, I see nothing else that really needs done just to push a release to get the ball rolling again.

However, if there's anything else you see that you want to take a try at to help, please feel free. Help would be awesome. I enjoy doing it on my own, but a few hands would really be christmas.

**Comment by hudmarc** [comment]
Hey, I've been looking to migrate my team (which uses Google Workspace) to this, currently we use the keepass desktop app but the non-techy team members struggle with it, so a web-based solution seems much more portable. Currently on the beta site I can add keepass as a trusted app but for some reason I can't view any shared folders or see anything shared with me. Is there some set-up step I'm missing with the GDrive integration?

<img width="838" height="351" alt="Image" src="https://github.com/user-attachments/assets/0742c559-9f33-432a-bdcc-3c50b663f508" />

**Comment by allohamora** [comment]
Hey everyone!

Since this issue has been abandoned and not fixed since 2024, I got tired of waiting and built [keeweb-lite](https://github.com/allohamora/keeweb-lite) - a lightweight KeeWeb reimplementation for my own needs, based on the same core packages (`argon2-browser`, `kdbxweb`) but with a modern stack, and support for **local files** and **Google Drive** strategies

Feel free to **use it as-is** or **fork it** and add whatever features you need for yourself. It's a great starting point, especially with AI agents
